File: src/main/java/com/dafttech/terra/engine/gui/containers/ContainerList.java
package com.dafttech.terra.engine.gui.containers;

import com.dafttech.terra.engine.AbstractScreen;
import com.dafttech.terra.engine.Vector2;
import com.dafttech.terra.engine.gui.GUIObject;

public class ContainerList extends GUIContainer {
    float distance = 2;

    public ContainerList(Vector2 p, Vector2 s) {
        super(p, s);
    }

    public ContainerList(Vector2 p, Vector2 s, float d) {
        super(p, s);
        distance = d;
    }

    @Override
    public void draw(AbstractScreen screen) {

        float y = size.y;

        for (GUIObject o : objects) {
            y -= o.size.y + distance;

            if (y > 0) {
                o.position = new Vector2(0, y);
                o.draw(screen);
            }
        }
    }
}

File: terra/src/main/scala/com/dafttech/terra/engine/gui/containers/ContainerList.scala
package com.dafttech.terra.engine.gui.containers

import com.dafttech.terra.engine.AbstractScreen
import com.dafttech.terra.engine.vector.Vector2d

class ContainerList(p: Vector2d, s: Vector2d) extends GUIContainer(p, s) {
  private var distance: Float = 2

  def this(p: Vector2d, s: Vector2d, d: Float) {
    this(p, s)
    distance = d
  }

  override def draw(screen: AbstractScreen): Unit = {
    var y: Float = size.y.toFloat

    for (o <- objects) {
      y -= o.size.y.toFloat + distance
      if (y > 0) {
        o.position = Vector2d(0, y)
        o.draw(screen)
      }
    }
  }
}